### PR TITLE
[EventHubs] fix tracing bug

### PR DESCRIPTION
**Pull request by swathipil**
fixes bug in: #27986

changes:
- removed duplicate tracing which was adding the `Azure.EventHubs.Message` span twice when sending a batch message/list of messages.
- adding the tracing info the uamqp/pyamqp.Message application_properties rather than event_data. Ensures that the diagnostic ID is actually being added on the outgoing Message, rather than the EventData after outgoing Message has already been created.
- Adding the FakeSpan class to `tests` folder to test telemetry.

TODO:
- follow-up issue created to address implementation so it behaves correctly in a separate PR: #28141

### CODE REVIEW

File: sdk/eventhub/azure-eventhub/tests/tracing_common.py [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+    Sequence[float],
+]
+Attributes = Optional[Dict[str, AttributeValue]]
+
+class FakeSpan(HttpSpanMixin, object):
+    # Keep a fake context of the current one
+    CONTEXT = []
+
+    def __init__(self, span=None, name="span", kind=None, *, links=[]):
+        # type: (Optional[Span], Optional[str]) -> None
```
Comment (kashifkhan) [inline sdk/eventhub/azure-eventhub/tests/tracing_common.py:27]: we should not have the comment style typing

File: sdk/eventhub/azure-eventhub/azure/eventhub/aio/_producer_async.py [hunk 2]
```diff
@@ -11,7 +11,7 @@
 from azure.core.tracing import AbstractSpan
 
 from .._common import EventData, EventDataBatch
-from .._producer import _set_partition_key, _set_trace_message
+from .._producer import _set_partition_key, _trace_batch_message
```
Comment (kashifkhan) [inline sdk/eventhub/azure-eventhub/azure/eventhub/aio/_producer_async.py:14]: is there a commit missing for `_trace_batch_message` ?

File: sdk/eventhub/azure-eventhub/azure/eventhub/aio/_producer_async.py [hunk 3]
```diff
@@ -210,20 +214,16 @@ def _wrap_eventdata(
...
             else:
                 if partition_key:
                     event_data = _set_partition_key(
                         event_data, partition_key, self._amqp_transport
                     )
-                event_data = _set_trace_message(event_data, span)
                 wrapper_event_data = EventDataBatch._from_batch(  # type: ignore  # pylint: disable=protected-access
                     event_data, self._amqp_transport, partition_key
                 )
+            _trace_batch_message(wrapper_event_data._message, span)
```
[the remark this answers is not in the document]
Comment (swathipil) [reply sdk/eventhub/azure-eventhub/azure/eventhub/aio/_producer_async.py]: no sorry, this should've been deleted. just removed it.

File: sdk/eventhub/azure-eventhub/azure/eventhub/aio/_producer_async.py (hunk 2 above)
Comment (swathipil) [reply sdk/eventhub/azure-eventhub/azure/eventhub/aio/_producer_async.py]: this was adding duplicate "Azure.EventHubs.Message" spans, so needed to be deleted. just removed.

File: sdk/eventhub/azure-eventhub/tests/tracing_common.py (hunk 1 above)
Comment (swathipil) [reply sdk/eventhub/azure-eventhub/tests/tracing_common.py:27]: yeah, I plan on fixing all this in the typing PR (which is incoming I promise :P)

**Comment by swathipil** [comment]
perf run: https://dev.azure.com/azure-sdk/internal/_build/results?buildId=2091776&view=logs&j=42ded549-05ee-5cdb-7ba7-7a948a0cc056&t=a103fa30-9714-51c6-dfba-9bd5390ed023

**Review by kashifkhan — APPROVED** [review]
LGTM!

**Pull request merged**